server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -307,7 +308,8 @@ static Custom fromXContent(XContentParser parser, String name) throws IOExceptio
         this.persistentSettings = persistentSettings;
         this.settings = Settings.builder().put(persistentSettings).put(transientSettings).build();
         this.hashesOfConsistentSettings = hashesOfConsistentSettings;
-        this.indices = Collections.unmodifiableMap(indices);
+        LinkedHashMap<String, IndexMetadata> linkedMap = new LinkedHashMap<>(indices);
```

Comment:
I see a lot of problems in this PR. Can you please work on a separate PR and raise with all concerns mitigated?
1. There is already an indices map in metadata. To add a new map just for a cat output doesn't make any sense.
2. Naming conventions are not followed. E.g. `linkedMap` doesn't portray the intent of the map. 
3. No functional tests to validate the functionality.
4. No backward compatibility version checks are present and I suspect that ti will fail for mixed versions.  No mixed cluster tests are present to validate the same.
5. I see use of string "null" instead of null and then a check on null is present in class Table, so I am not quite sure if this case will work.
6. No description in the PR to explain how the feature is intended to be built and use case(s) explaining it's benefit.

@gargharsh3134 , Thanks for raising the PR and contributing but unfortunately I believe that this PR doesn't follow standard PR practices and therefore, in my opinion it shouldn't be accepted. I encourage you to raise another PR with all concerns mitigated.

Tagging: @shwetathareja @Bukhtawar for their opinions and to take required action!

Reply:
@vikasvb90 Thanks for taking a look. I was aware of the above issues and just wanted a draft PR of the prototype that I did (mostly to give some high level context on the parameter and class level changes). I'm still working on the changes, will move this out of DRAFT state ONLY when all the concerns are addressed.  
Will also update the exact details in the issue which is already created.

Please let me know if I should still cancel this DRAFT PR.

Reply:
Got it! But I believe your initial commit also has issues and there was no description in the PR about the change. I am ok with keeping this in draft but please update PR description about the change you are planning to do. One uber concern I had was backward compatibility changes and tests. Please take care of the them in your PR if you haven't planned already.

Reply:
@vikasvb90 : @gargharsh3134  intended to get early feedback on the PR and will keep it in draft mode until it is ready to be reviewed by wider group.

Reply:
@shwetathareja @vikasvb90 I have removed the metadata changes from this PR (was trying to make indices metadata map ordered and deterministic, will followup on that in a separate PR along with addressing backward compatibility). Have added details around functional testing in the description (please check attached issue for more details).
Request you to take a look again (will add the required set of UTs and ITs, once changes are finalized).
Thanks!